### ISSUE TITLE
HostedUIOptions for Facebook and Google opens white blank screen for few seconds

### ISSUE THREAD

**Issue by shishir87** [aws-amplify/aws-sdk-android#1524]
Hi,

I am developing an android app in which user can log in via Facebook or Google.

When user selects one of these options, I am executing below lines of code.

==============
```
val hostedUi = HostedUIOptions.builder()
    .scopes("openid", "email")
    .identityProvider("Facebook”)// or “Google” depending upon user selection
    .build()


val signInUIOptions = SignInUIOptions.builder()
    .hostedUIOptions(hostedUi)
    .build()
AWSMobileClient.getInstance()
    .showSignIn(requireActivity(), signInUIOptions, object : Callback<UserStateDetails> {
        override fun onResult(details: UserStateDetails) {
         //user is logged in successfully
        }
        override fun onError(e: Exception) {
            //Log in is not successful
        }
    })
```
==============

If user has installed my app for first time then after executing above code I am getting the Facebook or Google login page. So, user can enter his Facebook or Google login credentials and complete his login procedure.

But if user log out from my app and try to login via Facebook or Google again, then a blank white screen gets opened for a moment and he gets logged in successfully.

Basically, I don’t want that white blank page which is visible for few seconds when user login second time.

**Comment by jpignata** [comment]
Hmm, is this a bug, @raphkim? I'd expect we'd need to open a browser and interact with the third party in this flow.

**Comment by raphkim** [comment]
Oops, I didn't realize we had a pending triage tag @jpignata

**Comment by TrekSoft** [comment]
Hi,

Sorry for the delay. This is an unfortunate side effect of the Google Chrome Custom tabs which are necessary to make OAuth authentication like this work with no known way of avoiding it.

If you want to require the user to sign in again or allow them to choose a different account, be sure to sign out with invalidateTokens set to true as described here: https://docs.amplify.aws/sdk/auth/hosted-ui/q/platform/android#sign-out-from-hostedui-1
